### PR TITLE
Home task4

### PR DESCRIPTION
**Pull request by elenagronskaya**

### CODE REVIEW

File: src/components/CourseForm/CourseForm.jsx [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+import getUsersSelector from '../../store/user/selectors';
+import {
+	createCourseThunk,
+	fillSelectedCourseThunk,
+} from '../../store/courses/thunk';
+import getCoursesSelector from '../../store/courses/selectors';
+import { setSelectedCourseProperty } from '../../store/courses/actions';
+import styles from './styles.module.scss';
+
+const CreateForm = () => {
```
Comment (AliaskarUrakovTR) [inline src/components/CourseForm/CourseForm.jsx]: Rename to CourseForm

File: src/components/CourseForm/CourseForm.jsx [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+		);
+	};
+
+	const createCourse = () => {
+		if (!validate()) {
+			alert('Please fill in all fields');
+			return;
+		}
+
+		store.dispatch(createCourseThunk({ ...selectedCourse }));
```
Comment (AliaskarUrakovTR) [inline src/components/CourseForm/CourseForm.jsx:54]: new course is not shown in the courses page after redirect, either save it in store in dispatch or refetch it again

File: src/components/CourseForm/CourseForm.jsx [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+		navigate(ROUTE_COURSES);
+	};
+
+	const setValueByName = (name, value) => {
+		store.dispatch(setSelectedCourseProperty(name, value));
+	};
+
+	const setValue = (event) => {
+		const name = event.target.name;
```
Comment (AliaskarUrakovTR) [inline src/components/CourseForm/CourseForm.jsx]: const { name, value } = event.target

File: src/components/CourseForm/CourseForm.jsx [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+					/>
+				</div>
+				<label className={styles.label}>
+					Description
+					<textarea
+						name='description'
+						className={styles.textarea}
+						placeholder='Enter description'
+						value={selectedCourse?.description || ''}
+						onChange={(event) => setValue(event)}
```
Comment (AliaskarUrakovTR) [inline src/components/CourseForm/CourseForm.jsx]: onChange={setValue}

File: src/components/Login/Login.jsx [hunk 5]
```diff
@@ -40,12 +33,14 @@ const Login = () => {
 	const handleSubmit = async (e) => {
```
Comment (AliaskarUrakovTR) [inline src/components/Login/Login.jsx]: no await, remove async

File: src/components/PrivateRoute/PrivateRoute.jsx [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+import { useEffect } from 'react';
+import { useSelector } from 'react-redux';
+import { Navigate } from 'react-router-dom';
+
+import { ROUTE_LOGIN } from '../../constants';
+import store from '../../store';
+import getUsersSelector from '../../store/user/selectors';
+import { getCurrentUserThunk } from '../../store/user/thunk';
+
+const PrivateRoute = ({ children, requiredRole }) => {
```
Comment (AliaskarUrakovTR) [inline src/components/PrivateRoute/PrivateRoute.jsx:11]: add prop types

File: src/components/PrivateRoute/PrivateRoute.jsx [hunk 7]
```diff
@@ -0,0 +1,30 @@
+import React from 'react';
```
Comment (AliaskarUrakovTR) [inline src/components/PrivateRoute/PrivateRoute.jsx]: remove

File: src/components/PrivateRoute/PrivateRoute.jsx [hunk 8]
```diff
@@ -0,0 +1,30 @@
...
+import { getCurrentUserThunk } from '../../store/user/thunk';
+
+const PrivateRoute = ({ children, requiredRole }) => {
+	const userData = useSelector(getUsersSelector);
+
+	useEffect(() => {
+		store.dispatch(getCurrentUserThunk());
+	}, []);
+
+	const { role } = userData;
```
Comment (AliaskarUrakovTR) [inline src/components/PrivateRoute/PrivateRoute.jsx]: empty line after

File: src/components/PrivateRoute/PrivateRoute.jsx [hunk 9]
```diff
@@ -0,0 +1,30 @@
...
+import { useSelector } from 'react-redux';
+import { Navigate } from 'react-router-dom';
+
+import { ROUTE_LOGIN } from '../../constants';
+import store from '../../store';
+import getUsersSelector from '../../store/user/selectors';
+import { getCurrentUserThunk } from '../../store/user/thunk';
+
+const PrivateRoute = ({ children, requiredRole }) => {
+	const userData = useSelector(getUsersSelector);
```
Comment (AliaskarUrakovTR) [inline src/components/PrivateRoute/PrivateRoute.jsx]: const { role, isAuth } = useSelector(getUsersSelector);

File: src/components/PrivateRoute/PrivateRoute.jsx [hunk 10]
```diff
@@ -0,0 +1,30 @@
...
+	const { role } = userData;
+	if (!userData.isAuth) {
+		return <Navigate to={ROUTE_LOGIN} />;
+	}
+
+	if (!requiredRole) {
+		return children;
+	}
+
+	return requiredRole === role ? children : <Navigate to={'/'} />;
```
Comment (AliaskarUrakovTR) [inline src/components/PrivateRoute/PrivateRoute.jsx]: it should navigate to /courses

File: src/store/authors/api.js [hunk 11]
```diff
@@ -0,0 +1,15 @@
+import axios from 'axios';
+import getHeaders from '../../api';
+import BASE_URL from '../../env';
```
Comment (AliaskarUrakovTR) [inline src/store/authors/api.js]: This is completely wrong. Learn how to work with environment variables in React
https://create-react-app.dev/docs/adding-custom-environment-variables/
https://dev.to/felixdusengimana/environment-variables-in-reactjs-139l

**Review by AliaskarUrakovTR — CHANGES_REQUESTED** [review]
Waiting for fixes

File: src/components/PrivateRoute/PrivateRoute.jsx [hunk 12]
```diff
@@ -0,0 +1,39 @@
...
+
+	if (!requiredRole) {
+		return children;
+	}
+
+	return requiredRole === role ? children : <Navigate to={'/courses'} />;
+};
+
+PrivateRoute.propTypes = {
+	children: PropTypes.object,
```
Comment (urakovaliaskar) [inline src/components/PrivateRoute/PrivateRoute.jsx:30]: children: PropTypes.node.isRequired

**Review by AliaskarUrakovTR — APPROVED** [review]

**Pull request merged**